### PR TITLE
Nav unification: consolidate color-sidebar-menu-hover-heading-background with non-"heading" version

### PR DESCRIPTION
**Pull request by mreishus**
#### Background

Compared to #47294, this is a slightly more complex change.  I definitely want @cpapazoglou to check this one out.

* This is considering the color of the background when hovering over sidebar items.
* Before we started the nav-unification project, calypso color schemes used `color-sidebar-menu-hover-background`.
* When we first added nav unification styles, we defined two variables, with the same color value:
  * `color-sidebar-menu-hover-heading-background`
    * This is a completely new variable to nav unification, not used in other places in calypso.  However, it seems to perform the same function as the `color-sidebar-menu-hover-background` variable in calypso.
    * It's basically used everywhere in nav unification.  So a theme must define the same value to `color-sidebar-menu-hover-heading-background` and `color-sidebar-menu-hover-background` to work with nav unification both on and off.
  * `color-sidebar-menu-hover-background`
    * This matches the calypso variable, but was only used in one corner case I couldn't recreate.  Something about the site switcher.  [Link to code](https://github.com/Automattic/wp-calypso/blob/f1841f5577afbaf8d98656aef42cf83f03f27ad8/client/my-sites/sidebar-unified/style.scss#L282-L284).
    * Its value in the default nav unification scheme is `#32373c`, which is a slightly darker version of the main color `#1a1e23`.  In pull request #47249, I added a `submenu-background` variable which is also defined `#32373c`.

#### Changes proposed in this Pull Request

* Let `color-sidebar-menu-hover-background` control the sidebar hover background in both calypso and calypso+nav-unification.
* For the corner case with the site switcher, change to `color-sidebar-submenu-background`, which is also `#32373c`.
* Remove the new `color-sidebar-menu-hover-heading-background` variable.

#### Benefits

* One less variable
* Less changes needed to calypso schemes when porting to nav-unification calypso.

#### Testing instructions

Visit calypso with nav unification turned on, http://calypso.localhost:3000/?flags=nav-unification. Hovering sidebars should still work.

### Reference Issues

* Main Issue #45435
  * Color Schemes Issue + Direction  #45675
    * Add the necessary wp-admin styles to support all color schemes available in Calypso #47040
      * Nav unification: Example of extending sakura to work on nav-unification #47292

### CODE REVIEW

**Comment by cpapazoglou** [comment]
> `color-sidebar-menu-hover-background`
This matches the calypso variable, but was only used in one corner case I couldn't recreate. Something about the site switcher. Link to code.

Is triggered when in site switcher you add a search term and press the down key

File: client/my-sites/sidebar-unified/style.scss [hunk 1]
```diff
@@ -9,9 +9,8 @@
 
 	--color-sidebar-background: #23282d;
 	--color-sidebar-background-rgb: 35, 40, 45;
-	--color-sidebar-menu-hover-background: #32373c;
-	--color-sidebar-menu-hover-background-rgb: rgb( #32373c );
-	--color-sidebar-menu-hover-heading-background: #1a1e23;
+	--color-sidebar-menu-hover-background: #1a1e23;
+	--color-sidebar-menu-hover-background-rgb: rgb( #1a1e23; );
```
Comment (cpapazoglou) [inline client/my-sites/sidebar-unified/style.scss:13]: Let's reuse `--color-sidebar-menu-hover-background` here:
--color-sidebar-menu-hover-background-rgb: rgb( var(--color-sidebar-menu-hover-background) );

**Review by cpapazoglou — APPROVED** [review]
Thanks @mreishus for wrangling this!
This change makes sense to me and looks good!

File: client/my-sites/sidebar-unified/style.scss (hunk 1 above)
Comment (mreishus) [reply client/my-sites/sidebar-unified/style.scss]: Fixed [fix/consolidate-nav-unification-vars-2 60fbc9452a] --color-sidebar-menu-hover-background-rgb computed via rgb function

**Pull request merged**